File: sdk/typescript/examples/agent-selection.ts
/**
 * Agent Selection Example
 *
 * Demonstrates how to list available agents, check their
 * capabilities, and select the best agents for a task.
 *
 * Usage:
 *   npx ts-node examples/agent-selection.ts
 */

import { createClient } from '@aragora/sdk';

async function main() {
  const client = createClient({
    baseUrl: process.env.ARAGORA_API_URL || 'https://api.aragora.ai',
    apiKey: process.env.ARAGORA_API_KEY,
  });

  // List all available agents
  console.log('=== Available Agents ===\n');
  const agents = await client.agents.list();

  for (const agent of agents.agents) {
    console.log(`${agent.name} (${agent.id})`);
    console.log(`  Provider: ${agent.provider}`);
    console.log(`  Capabilities: ${agent.capabilities?.join(', ') || 'general'}`);
    console.log(`  ELO Rating: ${agent.elo_rating || 'N/A'}`);
    console.log(`  Status: ${agent.status}`);
    console.log();
  }

  // Get recommendations for a specific task
  console.log('=== Agent Recommendations ===\n');
  const recommendations = await client.agents.recommend({
    task: 'Write and review a Python function for data validation',
    count: 3,
    criteria: {
      capabilities: ['code', 'reasoning'],
      minElo: 1200,
    },
  });

  console.log('Recommended agents for this task:');
  for (const rec of recommendations.recommendations) {
    console.log(`  ${rec.agent_id}: ${rec.reason} (score: ${rec.score})`);
  }

  // Create debate with recommended agents
  const debate = await client.debates.create({
    task: 'Write and review a Python function for data validation',
    agents: recommendations.recommendations.map((r) => r.agent_id),
    protocol: {
      rounds: 2,
      consensus: 'weighted',
    },
  });

  console.log(`\nDebate created with recommended agents: ${debate.debate_id}`);
}

main().catch(console.error);

File: sdk/python/examples/agent_selection.py
"""
Agent Selection Example

Demonstrates how to list available agents, view their capabilities,
check ELO rankings, and compare agent performance.

Usage:
    python examples/agent_selection.py

Environment:
    ARAGORA_API_KEY - Your API key
    ARAGORA_API_URL - API URL (default: https://api.aragora.ai)
"""

from __future__ import annotations

import os

from aragora_sdk import AragoraClient


def main() -> None:
    client = AragoraClient(
        base_url=os.environ.get("ARAGORA_API_URL", "https://api.aragora.ai"),
        api_key=os.environ.get("ARAGORA_API_KEY"),
    )

    # List all available agents
    print("=== Available Agents ===\n")
    agents = client.agents.list()

    for agent in agents.get("agents", []):
        name = agent.get("name", agent["id"])
        provider = agent.get("provider", "unknown")
        capabilities = ", ".join(agent.get("capabilities", ["general"]))
        elo = agent.get("elo_rating", "N/A")
        status = agent.get("status", "unknown")
        print(f"{name} ({agent['id']})")
        print(f"  Provider: {provider}")
        print(f"  Capabilities: {capabilities}")
        print(f"  ELO Rating: {elo}")
        print(f"  Status: {status}")
        print()

    # View the leaderboard
    print("=== Agent Leaderboard ===\n")
    leaderboard = client.agents.leaderboard()

    for i, entry in enumerate(leaderboard.get("rankings", [])[:10], 1):
        agent_id = entry.get("agent_id", "unknown")
        elo = entry.get("elo_rating", "N/A")
        wins = entry.get("wins", 0)
        losses = entry.get("losses", 0)
        print(f"  {i}. {agent_id} — ELO: {elo}, W/L: {wins}/{losses}")

    # Compare specific agents
    print("\n=== Agent Comparison ===\n")
    comparison = client.agents.compare(["claude", "gpt-4"])

    for agent_data in comparison.get("agents", []):
        agent_id = agent_data.get("agent_id", "unknown")
        print(f"{agent_id}:")
        stats = agent_data.get("stats", {})
        print(f"  Win rate: {stats.get('win_rate', 'N/A')}")
        print(f"  Avg confidence: {stats.get('avg_confidence', 'N/A')}")
        print(f"  Debates participated: {stats.get('debate_count', 'N/A')}")
        print()

    # Get detailed profile for a specific agent
    print("=== Agent Profile: claude ===\n")
    profile = client.agents.get_profile("claude")
    print(f"  Name: {profile.get('name', 'N/A')}")
    print(f"  Provider: {profile.get('provider', 'N/A')}")
    print(f"  ELO Rating: {profile.get('elo_rating', 'N/A')}")
    print(f"  Total debates: {profile.get('total_debates', 'N/A')}")
    print(f"  Specialties: {', '.join(profile.get('specialties', []))}")


if __name__ == "__main__":
    main()
